bliss/models/encoder.py

```diff
@@ -595,12 +572,11 @@ def tiled_map_estimate(self, image):
             "fluxes": tile_fluxes,
         }
 
-    def map_estimate(self, image):
-        slen = image.shape[-1]
-
-        if not self.pad_border_w_constant:
-            slen = slen - 2 * self.edge_padding
-
-        tile_estimate = self.tiled_map_estimate(image)
+    def map_estimate(self, slen, images):
+        # slen is size of the image without border padding
+        border_padding = (images.shape[-1] - slen) / 2
+        assert slen % self.tile_slen == 0, "incompatible image"
+        assert border_padding == self.border_padding, "incompatible border"
+        tile_estimate = self.tiled_map_estimate(images)
```

Comment:
Do you have to pass in the slen? Could we set

```
slen = images.shape[-1] - 2 * self.border_padding
```
and keep only the assert on line 578 (and remove the one on 579). 

I **think** everything that follows should go through.

Reply:
Yes, in theory we could do that like we discussed. But then I realized that if I use this same encoder for other images that were not the decoder ones, I would like to know that the border padding is consistent. Otherwise the locs prediction will be off because the slen is wrong